qa/tasks/scripts/create_all_pools_at_once.sh

```diff
@@ -22,17 +22,19 @@ function json_total_osds {
     ceph osd ls --format json | jq '. | length'
 }
 
+function power2() { echo "x=l($1)/l(2); scale=0; 2^((x+0.5)/1)" | bc -l; }
```

Comment:
Not clear what this function is supposed to do?

Reply:
Is this expected:
```
$ power2 18
16
```

Reply:
you have to look to whole code and just take part of it. This works.

Reply:
> Not clear what this function is supposed to do?

If the value of the above calculation is less than the value of ( OSD# ) / ( Size ), then the value is updated to the value of ( OSD# ) / ( Size ). This is to ensure even load / data distribution by allocating at least one Primary or Secondary PG to every OSD for every Pool.
The output value is then rounded to the nearest power of 2.
Tip: The nearest power of 2 provides a marginal improvement in efficiency of the CRUSH algorithm.
If the nearest power of 2 is more than 25% below the original value, the next higher power of 2 is used.

Reply:
does this mean something like this?
```
function power2() { i=1 ; while [ $1 -gt $i ] ; do i=$(( i << 1 )) ; done ; echo $i ; }
```

Reply:
Oh didn't know this OSD#/Size calculation and 25%, could you please add this in comments and put the reference to the doc where those magic numbers are taken.
Because power2 means for me power of 2, not something like taking into account of those 25%.

Reply:
example: your power of 2 for 22 is 16, 22 - 16 = 6, 6 is around 37% of 16, it is more than 25%, so the resulting value should be 32?

Reply:
well no, it should be always one closer to the given number.